packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/NodeStylePropertiesConfigurer.java

```diff
@@ -472,6 +481,37 @@ private Stream<UserColor> getColorsFromColorPalettesStream(VariableManager varia
                               .flatMap(EList::stream);
     }
 
+    private SelectDescription createIconSelectionField(Object feature) {
+        // @formatter:off
+        return SelectDescription.newSelectDescription("nodestyle.iconLabelSelector")
+                .idProvider(variableManager -> "nodestyle.iconLabelSelector")
+                .labelProvider(variableManager -> "Custom Icon")
+                .styleProvider(vm -> SelectStyle.newSelectStyle().showIcon(true).build())
+                .valueProvider(variableManager -> variableManager.get(VariableManager.SELF, NodeStyleDescription.class).map(NodeStyleDescription::getLabelIcon).orElse(EMPTY))
+                .optionsProvider(variableManager -> variableManager.get(IEditingContext.EDITING_CONTEXT, IEditingContext.class)
+                        .map(IEditingContext::getId)
+                        .map(this.customImageSearchService::getAvailableImages)
+                        .orElse(List.of())
+                )
+                .optionIdProvider(variableManager -> variableManager.get(SelectComponent.CANDIDATE_VARIABLE, CustomImageMetadata.class)
+                        .map(CustomImageMetadata::getId)
+                        .map(UUID::toString)
+                        .orElse(EMPTY))
+                .optionLabelProvider(variableManager -> variableManager.get(SelectComponent.CANDIDATE_VARIABLE, CustomImageMetadata.class)
+                        .map(CustomImageMetadata::getLabel)
+                        .orElse(EMPTY))
+                .optionIconURLProvider(variableManager -> variableManager.get(SelectComponent.CANDIDATE_VARIABLE, CustomImageMetadata.class)
+                        .map(CustomImageMetadata::getId)
```

Comment:
I think you can use `CustomImageMetadata#getURL` directly here, and avoid having to detect/parse/adapt a raw UUUID in the data fetcher.

Reply:
customImageSearchService::getAvailableImages return org.eclipse.sirius.components.view.emf.CustomImageMetadata where I don't have getURL.